From db62e4fe6e0812743b5fa366c3b766f1f24edc67 Mon Sep 17 00:00:00 2001
From: Yan Wang <r65094@freescale.com>
Date: Fri, 20 Feb 2009 14:51:47 +0800
Subject: [PATCH]     ENGR00107850 The system can not reboot using the "reboot" command

    This CR is required to resolve RNGC on-hold issue when hot reset
    It is the same issue with ENGR00086228

    Signed-off-by: Yan Wang <r65094@freescale.com>
---
 drivers/mxc/security/rng/rng_driver.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/drivers/mxc/security/rng/rng_driver.c b/drivers/mxc/security/rng/rng_driver.c
index d6a6bf6..bad9019 100644
--- a/drivers/mxc/security/rng/rng_driver.c
+++ b/drivers/mxc/security/rng/rng_driver.c
@@ -272,6 +272,10 @@ OS_DEV_INIT(rng_init)
 		/* Self Testing For RNG */
 		do {
 			RNG_CLEAR_ERR();
+
+			/* wait for Clearing Erring finished */
+			msleep(1);
+
 			RNG_UNMASK_ALL_INTERRUPTS();
 			RNG_SELF_TEST();
 #if !defined(FSL_HAVE_RNGA)
-- 
1.5.4.4

